rerun_py/src/dataframe.rs

```diff
@@ -0,0 +1,635 @@
+        match self {
+            Self::ComponentDescriptor(desc) => desc.0.into(),
+            Self::ComponentSelector(selector) => selector.0,
+        }
+    }
+}
+
+struct ComponentLike(re_sdk::ComponentName);
+
+impl FromPyObject<'_> for ComponentLike {
+    fn extract(component: &PyAny) -> PyResult<Self> {
+        if let Ok(component_str) = component.extract::<String>() {
+            Ok(Self(component_str.into()))
+        } else if let Ok(component_str) = component
+            .getattr("_BATCH_TYPE")
+            .and_then(|batch_type| batch_type.getattr("_ARROW_TYPE"))
+            .and_then(|arrow_type| arrow_type.getattr("_TYPE_NAME"))
+            .and_then(|type_name| type_name.extract::<String>())
+        {
+            Ok(Self(component_str.into()))
+        } else {
+            return Err(PyTypeError::new_err(
+                "ComponentLike input must be a string or Component class.",
+            ));
+        }
+    }
+}
+
+#[pyclass(frozen, name = "Schema")]
+#[derive(Clone)]
+pub struct PySchema {
+    // TODO(jleibs): This gets replaced with the new schema object
+    pub schema: Vec<ColumnDescriptor>,
+}
+
+#[pymethods]
+impl PySchema {
+    fn control_columns(&self) -> Vec<PyControlColumnDescriptor> {
+        self.schema
+            .iter()
+            .filter_map(|column| {
+                if let ColumnDescriptor::Control(col) = column {
+                    Some(col.clone().into())
+                } else {
+                    None
+                }
+            })
+            .collect()
+    }
+
+    fn time_columns(&self) -> Vec<PyTimeColumnDescriptor> {
+        self.schema
+            .iter()
+            .filter_map(|column| {
+                if let ColumnDescriptor::Time(col) = column {
+                    Some(col.clone().into())
+                } else {
+                    None
+                }
+            })
+            .collect()
+    }
+
+    fn component_columns(&self) -> Vec<PyComponentColumnDescriptor> {
+        self.schema
+            .iter()
+            .filter_map(|column| {
+                if let ColumnDescriptor::Component(col) = column {
+                    Some(col.clone().into())
+                } else {
+                    None
+                }
+            })
+            .collect()
+    }
+
+    fn column_for(
+        &self,
+        entity_path: &str,
+        component: ComponentLike,
+    ) -> Option<PyComponentColumnDescriptor> {
+        let entity_path: EntityPath = entity_path.into();
+
+        self.schema.iter().find_map(|col| {
+            if let ColumnDescriptor::Component(col) = col {
+                if col.matches(&entity_path, &component.0) {
+                    return Some(col.clone().into());
+                }
+            }
+            None
+        })
+    }
+}
+
+#[pyclass(name = "Recording")]
+pub struct PyRecording {
+    store: ChunkStore,
+    cache: re_dataframe2::external::re_query::Caches,
```

Comment:
For a more publicly-stable version of this:
```suggestion
    cache: re_dataframe2::QueryCache,